src/error.rs

```diff
@@ -33,12 +33,29 @@ impl NeedSize {
     }
 }
 
+/// Magic needed to frame message
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct NeedMagic {
+    magic: Vec<u8>,
+}
+
+impl NeedMagic where {
+    /// Create new [NeedSize] from magic bytes 
+    #[inline]
+    pub fn new(magic: &[u8]) -> Self {
+        Self { magic: magic.to_vec() }
+    }
+
+}
+
 /// Deku errors
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[non_exhaustive]
 pub enum DekuError {
     /// Parsing error when reading
     Incomplete(NeedSize),
+    /// Failed to find magic
+    Framing(NeedMagic),
```

Comment:
would prefer the name relate to magic, unless we plan on expanding this to be an enum with different variants